src/FakeItEasy/Configuration/AfterMustHaveHappenedConfiguration.cs

```diff
@@ -17,11 +17,38 @@ public AfterMustHaveHappenedConfiguration(FakeManager fakeManager, ICallMatcher
             this.repeatConstraint = repeatConstraint;
         }
 
-        public void InOrder(ISequentialCallContext context)
+        /// <summary>
+        /// Checks that the asserted call happened in order relative to others in the assertion chain.
+        /// </summary>
+        /// <param name="nextAssertion">An assertion describing the next call that should occur.</param>
+        /// <returns>An object that can be used to assert that a following call was made in the expected order.</returns>
+        /// <exception cref="ExpectationException">The call was not made in the expected order.</exception>
+        public IOrderableCallAssertion Then(IAfterMustHaveHappenedConfiguration nextAssertion)
         {
-            Guard.AgainstNull(context, "context");
+            var context = ServiceLocator.Current.Resolve<SequentialCallContext>();
+            this.CheckCallHappenedInOrder(context);
+            return new SubsequentOrderedCallAssertion(context).Then(nextAssertion);
+        }
 
+        private void CheckCallHappenedInOrder(SequentialCallContext context)
+        {
             context.CheckNextCall(this.fakeManager, this.matcher.Matches, this.callDescription, this.repeatConstraint);
         }
+
+        private class SubsequentOrderedCallAssertion : IOrderableCallAssertion
+        {
+            private readonly SequentialCallContext context;
+
+            public SubsequentOrderedCallAssertion(SequentialCallContext context)
+            {
+                this.context = context;
+            }
+
+            public IOrderableCallAssertion Then(IAfterMustHaveHappenedConfiguration nextCall)
+            {
+                ((AfterMustHaveHappenedConfiguration)nextCall).CheckCallHappenedInOrder(this.context);

```

Comment:
> Your idea of returning an abstract class with protected constructor (in order to allow clients to extend the class, I assume) should resolve the problem.

Actually, no, I didn't intend to allow the client to extend the class (which wouldn't make much sense). I meant to say an _internal_ constructor, not a protected one.



> Your suggested names are fine, but I'd considered a different tack. One reason for the class to be exposed is to distinguish it from the result of the `Then` to avoid the nested `Then`s. While thinking of that aspect, I thought of `UnorderedCallAssertion`, which a raw `MustHaveHappened` is. It's only by wrapping one in a `Then` or using the `Then` method on one that we come up with an ordered assertion. Then the class that I've called `SubsequentOrderedCallAssertion` could drop the `Subsequent`: once we start returning from `Then`, we've imposed an ordering.

Sounds good!